frontend/src/metabase/parameters/components/ValuesSourceModal/ValuesSourceTypeModal.tsx

```diff
@@ -375,29 +393,67 @@ const getSupportedFields = (question: Question) => {
   return fields.filter(field => field.isString());
 };
 
+const getSourceTypeOptions = (
+  parameter: ParameterWithTemplateTagTarget,
+): RadioOption<ValuesSourceType>[] => {
+  return [
+    ...(parameter.hasVariableTemplateTagTarget
+      ? []
+      : [{ name: t`From connected fields`, value: null }]),
+    { name: t`From another model or question`, value: "card" },
+    { name: t`Custom list`, value: "static-list" },
+  ];
+};
+
+interface UseParameterValuesOpts {
+  parameter: Parameter;
+  sourceType: ValuesSourceType;
+  sourceConfig: ValuesSourceConfig;
+  onFetchParameterValues: (
+    opts: FetchParameterValuesOpts,
+  ) => Promise<ParameterValues>;
+}
+
+const useParameterValues = ({
+  parameter: initialParameter,
+  sourceType,
+  sourceConfig,
+  onFetchParameterValues,
+}: UseParameterValuesOpts) => {
+  const [values, setValues] = useState<ParameterValue[]>();
+  const handleFetchValues = useSafeAsyncFunction(onFetchParameterValues);
+  const isValidSource = isValidSourceConfig(sourceType, sourceConfig);
+
+  const parameter = useMemo(
+    () => ({
+      ...initialParameter,
+      values_source_type: sourceType,
+      values_source_config: sourceConfig,
+    }),
+    [initialParameter, sourceType, sourceConfig],
+  );
+
+  useEffect(() => {
+    if (isValidSource) {
```

Comment:
Here we skip calls to API for intermediate values config, e.g. when you selected a question but haven't selected a field yet.